### PR TITLE
Adds AppEngine.Devices send data functionality

### PR DESCRIPTION
**Pull request by szakhlypa**

### CODE REVIEW

File: lib/astarte/client/appengine/devices.ex [hunk 1]
```diff
@@ -46,6 +46,48 @@ defmodule Astarte.Client.AppEngine.Devices do
     end
   end
 
+  def get_property_path_data(%AppEngine{} = client, device_id, interface, path, query_opts \\ [])
```
Comment (rbino) [inline lib/astarte/client/appengine/devices.ex:49]: What about merging this with the previous one by passing the path as opt (and moving query opts inside the opts) like this:
```
def get_properties_data(%AppEngine{} = client, device_id, interface, opts \\ []) do
  [...]
  request_path =
    if path = opts[:path] do
      "devices/#{device_id}/interfaces/#{interface}#{path}"
    else
      "devices/#{device_id}/interfaces/#{interface}
    end
  query = Keyword.get(opts, :query, [])
  ...
end

Comment (rbino) [reply lib/astarte/client/appengine/devices.ex:49]: Note that I expect the leading slash to be included in the path, so I don't add it after `interface`

File: lib/astarte/client/appengine/devices.ex [hunk 2]
```diff
@@ -46,6 +46,48 @@ defmodule Astarte.Client.AppEngine.Devices do
...
+    with {:ok, %Tesla.Env{} = result} <- Tesla.get(tesla_client, request_path, query: query_opts) do
+      if result.status == 200 do
+        {:ok, result.body}
+      else
+        {:error, %APIError{status: result.status, response: result.body}}
+      end
+    end
+  end
+
+  def set_property_path_data(%AppEngine{} = client, device_id, interface, path, data)
```
Comment (rbino) [inline lib/astarte/client/appengine/devices.ex:63]: I'd just call this `set_property`

File: lib/astarte/client/appengine/devices.ex [hunk 3]
```diff
@@ -46,6 +46,48 @@ defmodule Astarte.Client.AppEngine.Devices do
...
+    with {:ok, %Tesla.Env{} = result} <- Tesla.put(tesla_client, request_path, %{data: data}) do
+      if result.status == 200 do
+        :ok
+      else
+        {:error, %APIError{status: result.status, response: result.body}}
+      end
+    end
+  end
+
+  def unset_property_path_data(%AppEngine{} = client, device_id, interface, path)
```
Comment (rbino) [inline lib/astarte/client/appengine/devices.ex:77]: As above, `unset_property`

File: lib/astarte/client/appengine/devices.ex [hunk 4]
```diff
@@ -59,4 +101,18 @@ defmodule Astarte.Client.AppEngine.Devices do
       end
     end
   end
+
+  def send_datastream_data(%AppEngine{} = client, device_id, interface, path, data)
```
Comment (rbino) [inline lib/astarte/client/appengine/devices.ex:105]: Call this `send_datastream`

**Review by rbino — CHANGES_REQUESTED** [review]

**Review by rbino — APPROVED** [review]

**Review by bettio — APPROVED** [review]

**Pull request merged**